src/guiChatConsole.cpp

```diff
@@ -611,7 +611,7 @@ bool GUIChatConsole::OnEvent(const SEvent& event)
 			prompt.nickCompletion(names, backwards);
 			return true;
 		}
-		else if(event.KeyInput.Char != 0 && !event.KeyInput.Control)
+		else if(isprint(event.KeyInput.Char) && !event.KeyInput.Control)
```

Comment:
It looks like it's converting event.KeyInput.Char (which isn't a real wchar_t apparently) from UTF-8 to a codepoint.